docs/dev/guidelines/server-tests.rst

```diff
@@ -0,0 +1,175 @@
+************
+Server Tests
+************
+
+This section covers recommended practices for writing Artemis server tests. If you want to write tests for Artemis programming exercises to test students' submissions, check out `this <https://confluence.ase.in.tum.de/display/ArTEMiS/Best+Practices+for+writing+Java+Programming+Exercise+Tests+in+Artemis>`__.
+
+0. General testing tips
+========================
+Use appropriate and descriptive names for test cases so developers can easily understand what you test without looking deeper into it.
+To increase readability, prefix variable names with ``actual`` and ``expected`` instead of naming them ``int a``, ``double b``, ``String c``.
```

Comment:
Maybe elaborate a bit on that since IMO there are two points here:
- prefixing variables with `actual` and `expected`
- having descriptive names

The first point is clear but maybe elaborate on the second instead of only giving examples

Reply:
Makes sense, I seperated it into two sentences :+1: